worker/worker.go

```diff
@@ -45,13 +45,30 @@ func (w *Pool) StartWorkers() {
 				batchReadTime := time.Now()
 				//@TODO - Should add integration tests to prove that the worker receives the same message that it produced, on the delivery channel it created
 
+				eventCounts := make(map[string]int)
+				for _, e := range request.GetEvents() {
```

Comment:
This iteration can be avoided if this count and the respective metric is produced in the publisher.kafka directly, The producer already iterates over the events.

Reply:
@ramey ^^